dlt/common/typing.py

```diff
@@ -119,7 +120,34 @@ def asstr(self, verbosity: int = 0) -> str:
         ...
 
 
-def extract_type_if_modifier(t: Type[Any]) -> Type[Any]:
+def get_type_name(t: Type[Any]) -> str:
+    """Returns a human-friendly name of type `t`"""
+    if name := getattr(t, "__name__", None):
+        return name  # type: ignore[no-any-return]
+    t = get_origin(t) or t
+    if name := getattr(t, "__name__", None):
+        return name  # type: ignore[no-any-return]
+    return str(t)
+
+
+def is_callable_type(hint: Type[Any]) -> bool:
+    """Checks if `hint` is callable: a function or callable class. This function does not descend
+    into type arguments ie. if Union, Literal or NewType contain callables, those are ignored"""
+    if get_origin(hint) is get_origin(Callable):
```

Comment:
Also can't we directly compare it to `Callable` like `get_origin(hint) is Callable:`

Reply:
do you retract part of your code above?